draft-ietf-jsonpath-base.md

```diff
@@ -1443,17 +1443,26 @@ function-argument   = literal /
                       function-expr
 ~~~
 
-Any function expressions in a query must be well formed (by conforming to the above ABNF)
-and well typed,
+Any function expressions in a query must be well formed (by conforming to the above ABNF),
+supported, and well typed,
 otherwise the JSONPath implementation MUST raise an error
 (see {{synsem-overview}}).
+
+For a function expression to be supported, each function extension involved in the expression
+must be supported.
```

Comment:
This still feels weird to me.  We have

- function expression
- function extension
- expression

```suggestion
For a filter expression to be supported, each function extension involved in the expression
must be supported.
```

I think this is what we mean.  We call the entire expression a "filter expression."  We don't define a "function expression" anywhere that I'm aware (although the function syntax itself is `function-expr` which may be a misnomer).

Reply:
`function-expr` is defined in the ABNF as you observe (and includes expressions involving nested function calls). That ABNF section is immediately followed by "Any function expressions in a query [...]". §2.4.1 is entitled "Type System for Function Expressions" and §2.4.3 "Well-typedness of Function Expressions". So I think the term is well defined and can and should be used here in preference to "filter expression".

Reply:
What you have is weird and feels redundant and a bit confusing.

Reply:
I don't follow. This sentence defines function expression support (required for validity by the previous paragraph) with function extension support. Why is this redundant? What's confusing about it?

Reply:
Again, support is not a function of the query, so we cannot make this statement.